src/Raven.Server/Documents/PeriodicBackup/BackupHistory/BackupHistoryStorage.cs

```diff
@@ -0,0 +1,612 @@
+            Console.WriteLine();
+            Console.WriteLine($"Latest full backup created at: {new DateTime(latestFullBackupCreatedAtTicks):O}.");
+            foreach (var fullBackupSeekResult in backupHistoryTable.SeekForwardFromPrefix(BackupHistoryTableSchema.Indexes[ByBackupKindSlice], fullBackupStartSlice, fullBackupPrefixSlice, skip: 0))
+            {
+                var fullBackupCreatedAtTicks = Bits.SwapBytes(*(long*)fullBackupSeekResult.Result.Reader.Read((int)BackupHistorySchema.BackupHistoryColumns.CreatedAtTicks, out _));
+                // If the full backup is newer than the cutoff time, we can stop searching as entries are sorted chronologically
+                // If the full backup is the latest one, we won't delete it
+                if (fullBackupCreatedAtTicks >= cutoffTime.Ticks || fullBackupCreatedAtTicks == latestFullBackupCreatedAtTicks)
+                {
+                    Console.WriteLine($"Full backup created at: {new DateTime(fullBackupCreatedAtTicks):O} is newer than the cutoff time: {cutoffTime:O}, fullBackupCreatedAtTicks == latestFullBackupCreatedAtTicks: {fullBackupCreatedAtTicks == latestFullBackupCreatedAtTicks}.");
+                    break;
+                }
+
+                Console.WriteLine($"Full backup created at: {new DateTime(fullBackupCreatedAtTicks):O} is older than the cutoff time: {cutoffTime:O}, fullBackupCreatedAtTicks == latestFullBackupCreatedAtTicks: {fullBackupCreatedAtTicks == latestFullBackupCreatedAtTicks}.");
+
+                var fullBackupResultDetailsId = -1L;
+                if (TryGetBackupResultDetailsReader(tx, databaseName, taskId, fullBackupCreatedAtTicks, out var fullBackupReader))
+                    fullBackupResultDetailsId = fullBackupReader.Id;
+
+                result.Add((fullBackupSeekResult.Result.Reader.Id, fullBackupResultDetailsId));
+
+                using (BackupHistorySchema.GetByFullBackupIdIndexKeyPrefix(tx.InnerTransaction.Allocator, databaseName, taskId, fullBackupCreatedAtTicks, BackupKind.Incremental, out var prefixSlice))
+                using (BackupHistorySchema.GetByFullBackupIdIndexKey(tx.InnerTransaction.Allocator, databaseName, taskId, fullBackupCreatedAtTicks, BackupKind.Incremental, createdAtTicks: 1, out var firstSlice))
+                {
+                    foreach (var incrementalBackupSeekResult in backupHistoryTable.SeekForwardFromPrefix(BackupHistoryTableSchema.Indexes[ByFullBackupIdSlice], firstSlice, prefixSlice, skip: 0))
+                    {
+                        var incrementalBackupCreatedAtTicks = Bits.SwapBytes(*(long*)incrementalBackupSeekResult.Result.Reader.Read((int)BackupHistorySchema.BackupHistoryColumns.CreatedAtTicks, out _));
+                        Console.WriteLine($"Incremental backup created at: {new DateTime(incrementalBackupCreatedAtTicks):O}.");
```

Comment:
I wonder if those console writes are debug only or we should really put something to the logs? Not here probably but maybe the caller should log something upon removal of old backup history entries?

Reply:
Great